knowledge-graph/src/test/scala/io/renku/knowledgegraph/projects/details/ProjectJsonLDEncoderSpec.scala

```diff
@@ -49,14 +49,14 @@ class ProjectJsonLDEncoderSpec extends AnyWordSpec with should.Matchers with Sca
         resourceId   <- cursor.downEntityId.as[projects.ResourceId]
         identifier   <- cursor.downField(schema / "identifier").as[projects.GitLabId]
         path         <- cursor.downField(renku / "projectPath").as[projects.Path]
-        name         <- cursor.downField(schema / "name").as[projects.Name]
+        name         <- cursor.downField(schema / "name").as[Option[projects.Name]].flatMap(projects.Name.failIfNone)
```

Comment:
Not sure if I understand: if we do `failIfNone` at the end - wouldn't it make sense to use the previous decoder, that fails by itself on a blank value? (due to the NonBlank constraint)?

Reply:
Hmm, that's a good question. I somehow thought the previous decoder wouldn't be needed anymore due to the fact we know these fields may be coming with blank values. We could leave the old decoders along with the new ones producing Options, however, we might go into potential issues where we decode to Option as there will be a bit of ambiguity for the compiler. You know what I mean, the compiler will need to choose between
* the new decoder producing Option[TT]
* or the old decoder producing TT combined with the standard optionalDecoder.
I somehow find myself leaning towards not keeping the old decoder on these few types and using the `failIfNone` eventually if those few places.
Any more thoughts?

Reply:
Ah I see, I somehow missed that the old ones were removed. I was hoping the compiler is fine if certain types only provide decoders for `Option[TT]` and others do for `T`.  But you're right, it could be a problem! I'm anyways totally fine wtih the final flatMap in a few cases